collection_requirements.rst

```diff
@@ -114,10 +114,12 @@ For collections under ansible-collections the repository SHOULD be named ``NAMES
 
 For collections created for working with a particular entity, they should contain the entity name, for example ``community.mysql``.
 
-We should avoid:
+For corporate maintained collections, the repository can be named ``COMPANY_NAME.PRODUCT_NAME``, for example ``ibm.db2``.
 
-* too long FQCN names: try to make it compact but clear
-* nested namespaces if this is not really necessary, especially namespaces with repetitive parts
+We should avoid FQCN / repository names:
```

Comment:
Did I originally write this?
`community.mysql` is FQCN, and the preferred format

Is this bit actually about *collection* & *namespace* naming?

Reply:
@gundalow, no, you didn't.
yes, `community.mysql` is FQCN and repo name as far as i know.
```Is this bit actually about collection & namespace naming?``` yes, because they not always but often correspond to each other. We could split it into 2 paragraphs but wouldn't it be more confusing. maybe not, any thoughts?

Reply:
the paragraph was originally written by you:) but not the marked part (not lines 119-120). then i add several rows to it, then you merged. Now we are trying to make it better in accordance to @bmillemathias 's suggestions in the previous PR